src/atap_widgets/concordance.py

```diff
@@ -506,3 +796,162 @@ def get_reversed_words(s: pd.Series):
             raise ValueError(
                 "Invalid context option: should be 'left_context' or 'right_context'"
             )
+
+class DataIngestWidget:
+    """
+    New widget to go with pandas stlying approach
+    """
+
+    def __init__(self, df: pd.DataFrame, ungrouped_data: pd.DataFrame,results_per_page: int = 20, largest_line_length:int = 200):
+        """
+        Args:
+            df: DataFrame containing texts, as returned by prepare_text_df()
+            results_per_page: How many search results to show at a time
+        """
+        self.data = df #prepped df 
+        self.ungrouped_data = ungrouped_data
+        self.user_column_list = self._get_user_columns()
+        self.largest_line_length = largest_line_length
+        self.results_per_page = results_per_page
+        self.search_table = ConcordanceTable(df=self.data,stylingOn = True,ungrouped_data = self.ungrouped_data)
+
+    def _get_user_columns(self):
+        user_column_list = [col for col in self.ungrouped_data.columns if col not in ["row","chunk","spacy_doc"]]
+        user_column_list.insert(0,"NotSet")
+        return user_column_list
+    def show(self):
+        """
+        Display the interactive widget with pandas styling of the results from within ConcordanceTable
+        """
+
+        def display_results(page: int, **kwargs):
+            if not kwargs["keyword"]:
+                return None
+            for attr, value in kwargs.items():
+                setattr(self.search_table, attr, value)
+
+            try:
+                html = self.search_table._get_html2(page=page) 
+                if self.search_table.is_regex_valid():
+                    results = self.search_table._get_results()
+                    # Need at least one page
+                    n_pages = max(
+                        self.search_table._get_total_pages(n_results=len(results)), 1
+                    )
+                    page_input.max = n_pages
+            except NoResultsError:
+                n_pages = 1
+
+            display(ipywidgets.HTML(html))
+            return html
+
+        keyword_input = ipywidgets.Text(description="Keyword(s):")
+        regex_toggle_input = ipywidgets.Checkbox(
+            value=False,
+            description="Enable regular expressions",
+            disabled=False,
+            style={"description_width": "initial"},
+        )
+        ignore_case_input = ipywidgets.Checkbox(
+            value=True, description="Ignore case", disabled=False
+        )
+        whole_word_input = ipywidgets.Checkbox(
+            value=False,
+            description="Match whole words",
+            disabled=False,
+            style={"description_width": "initial"},
+        )
+        page_input = ipywidgets.BoundedIntText(
+            value=1,
+            min=1,
+            max=1,
+            step=1,
+            description="Page:",
+        )
+        window_width_input = ipywidgets.BoundedIntText(
```

Comment:
Can this be set to a default width of 50 and a step size of 1 now? It makes for a much nicer looking default.

Reply:
done